src/main/scala/com/cloudant/clouseau/Utils.scala

```diff
@@ -29,4 +29,55 @@ object Utils {
     new BytesRef(string)
   }
 
+  /**
+   * The function `elementWise` maps over elements of a list and applies a partial function `pf`
```

Comment:
Based on the description, "elementWise" may not be the name that suits the definition the most.  If this is intended to be refine a generic list to a specific list, that is more like a coercion.  Something like `coerceList`.

Reply:
how about?

- `refineElementsType`
- `refineInnerType`
- `ensureInnerType`
- `verifyInnerType`
- `ensureElementsType`
- `coerceElementsType`
- `coerceInnerType`

Reply:
Thanks for the suggestions — I like `ensureElementsType`.